spyder/api/plugins.py

```diff
@@ -533,3 +557,874 @@ def set_ancestor(self, ancestor):
         Needed to update the ancestor/parent of child widgets when undocking.
         """
         pass
+
+
+##############################################################################
+#
+# New API, to migrate plugins one by one and make it easier to test changes
+#
+##############################################################################
+class SpyderPluginV2(QObject):
+    """
+    A Spyder plugin to enhance functionality without a dockable widget.
+
+    If you want to create a plugin that adds a new panel, use
+    SpyderDockableWidget.
+    """
+
+    # --- API: Mandatory attributes ------------------------------------------
+    # ------------------------------------------------------------------------
+    # Name of the plugin that will be used to refer plugins.
+    # This name must be unique and will only be loaded once.
+    NAME = None
+
+    # --- API: Optional attributes ------------------------------------------
+    # ------------------------------------------------------------------------
+    # List of required plugin dependencies names
+    # Example: ['plots', 'ipython_console', ...]
+    REQUIRES = None
+
+    # List of required plugin dependencies names
+    # Example: ['plots', 'ipython_console', ...]
+    OPTIONAL = None
+
+    # This is the a widget (or any non widget class) to store behavior and
+    # actions. Subclass SpyderWigdetMixin
+    WIDGET_CLASS = None
```

Comment:
I don't get this. Why could this be needed for SpyderPlugin's? Besides, the comment is confusing because I don't understand what you mean by `Subclass SpyderWigdetMixin`.

Reply:
The Working Directory plugin creates a New widget, a Toolbar. In those cases it makes sense to put that widget here. If a Plugin is creating "several" non dock widgets, a status bar widgets, a menu, a new toolbar, those widgets could (or should be grouped) within some "parent"widget.

Reply:
It would help if the comment describes what is done with the class stored in `WIDGET_CLASS`.